### PR TITLE
Document ImagePullPolicy

### LINKED ISSUE #305: Document ImagePullPolicy
Add docs about ImagePullPolicy, options and defaults

### PR DESCRIPTION
**Pull request by chrisjs**
Resolves #305

### CODE REVIEW

File: spring-cloud-dataflow-server-kubernetes-docs/src/main/asciidoc/configuration.adoc [hunk 1]
```diff
@@ -284,6 +284,24 @@ With the Service Account name, set the `SPRING_CLOUD_DEPLOYER_KUBERNETES_DEPLOYM
...
+
+An Image Pull Policy can be configured for all deployed applications, for example:
+
+```
+        env:
+        - name: SPRING_CLOUD_DEPLOYER_KUBERNETES_DEPLOYMENT_IMAGE_PULL_POLICY
+          value: Always
+```
+
+NOTE: An Image Pull Policy defined on a per application basis takes precedence over this configuration setting.
```
Comment (sabbyanandan) [inline spring-cloud-dataflow-server-kubernetes-docs/src/main/asciidoc/configuration.adoc]: Perhaps you could clarify it to explicitly include the stream/task deployment in this statement. Maybe: "... per application basis while deploying the stream or task ..."

File: spring-cloud-dataflow-server-kubernetes-docs/src/main/asciidoc/getting-started.adoc [hunk 2]
```diff
@@ -879,3 +879,20 @@ Replacing <app> with the name of your app to be deployed using the provided Serv
 
 NOTE: A Service Account name defined on a per application basis take precedence over a configured server `SPRING_CLOUD_DEPLOYER_KUBERNETES_DEPLOYMENT_SERVICE_ACCOUNT_NAME` environment variable.
 
+=== Image Pull Policy
```
Comment (sabbyanandan) [inline spring-cloud-dataflow-server-kubernetes-docs/src/main/asciidoc/getting-started.adoc:882]: Possible to reuse and not duplicate the property definitions? Even section headers would be a dupe.

Comment (chrisjs) [reply spring-cloud-dataflow-server-kubernetes-docs/src/main/asciidoc/getting-started.adoc:882]: agreed on the duplication part -- as with a couple others, if we support configuring at the property level and a server config level it gets documented in both places to show both configuration styles. maybe we should revisit how the documentation for properties is structured in general to consolidate?

**Comment by chrisjs** [comment]
merged in 22b9d

**Pull request closed without merging**